doc/metadata.md

```diff
@@ -72,22 +69,80 @@
         chunk chunks[];
     }
 
+    // this must be a fixed size so we can binary search over it easily
     struct inode {
         u64 ino;
         inode_type type;
+        u32 uid;
+        u32 gid;
+        u16 mode;
+        u64 mtime, atime, ctime; /* seems like we should require these? */
         union {
-            // directory
+            // fifo
+            struct {
+                /* nothing additional about fifos */
+            },
+
+            // chr
+            struct {
+                dev_t major;
+                dev_t minor;
+            },
+
+            // dir
             struct {
-                u64 offset;
+                u64 dir_offset;
             },
 
-            // file
+            // blk; do we even want these? seems like maybe not since they're
+            // system specific.
+            struct {
+                dev_t major;
+                dev_t minor;
+            },
+
+            // reg
             struct {
                 u64 file_size; /* total file size */
-                u64 offest;
+                u64 file_offest;
+            },
+
+            // lnk
+            struct {
+                char target[PATH_MAX];
```

Comment:
do you think it makes sense to move `target` to the additional inode metadata section?  Depending on `PATH_MAX` it could waste some memory also when it is not needed (and possibly could it be a variable length string)?

Reply:
Yeah, that sounds reasonable.